### PR TITLE
Modernize Project and Add Support for FIPS Systems

### LINKED ISSUE #9: FIPS preventing remote connection
# 🐛 Bug Report
Tagging @felddy and @dav3r 
```
yavin | FAILED! => {

    "changed": true,

    "msg": "non-zero return code",

    "rc": 1,

    "stderr": "Shared connection to yavin closed.\r\n",

    "stderr_lines": [

        "Shared connection to yavin closed."

    ],

    "stdout": "Traceback (most recent call last):\r\n  File \"/home/gollam/.ansible/tmp/ansible-tmp-1593119061.2345998-36805-301601859236/ioc_scanner.py\", line 130, in <module>\r\n    sys.exit(main())\r\n  File \"/home/gollam/.ansible/tmp/ansible-tmp-1593119061.2345998-36805-301601859236/ioc_scanner.py\", line 102, in main\r\n    hashes = hash_file(file)\r\n  File \"/home/gollam/.ansible/tmp/ansible-tmp-1593119061.2345998-36805-301601859236/ioc_scanner.py\", line 48, in hash_file\r\n    hash_md5 = hashlib.md5()  # nosec\r\nValueError: error:060800A3:digital envelope routines:EVP_DigestInit_ex:disabled for fips\r\n",

    "stdout_lines": [

        "Traceback (most recent call last):",

        "  File \"/home/gollam/.ansible/tmp/ansible-tmp-1593119061.2345998-36805-301601859236/ioc_scanner.py\", line 130, in <module>",

        "    sys.exit(main())",

        "  File \"/home/gollam/.ansible/tmp/ansible-tmp-1593119061.2345998-36805-301601859236/ioc_scanner.py\", line 102, in main",

        "    hashes = hash_file(file)",

        "  File \"/home/gollam/.ansible/tmp/ansible-tmp-1593119061.2345998-36805-301601859236/ioc_scanner.py\", line 48, in hash_file",

        "    hash_md5 = hashlib.md5()  # nosec",

        "ValueError: error:060800A3:digital envelope routines:EVP_DigestInit_ex:disabled for fips"

    ]

}
```
## To Reproduce

Run the following:
`ansible -i hosts-file all -m script -e "--file md5sums.txt"  -a ./src/ioc_scan/ioc_scanner.py  --become
`
## Expected behavior

Expect no errors

## Any helpful log output

Workaround:

Edit this line in ioc_scanner.py

`hash_md5 = hashlib.md5(usedforsecurity=False) # nosec
`

```bash

```

### PR DESCRIPTION
**Pull request by mcdonnnj**
# 

## 🗣 Description
This PR updates this project to transition from TravisCI to GitHub Actions, adds an appropriate Lineage configuration file, and adds support for FIPS systems to resolve #9 . There are also some updates to code logic to make it more easily extensible for new hash types and to DRY out some code.

## 💭 Motivation and Context
@bra1ncramp pointed out that on a systems in FIPS mode the project as it is will not correctly run. As part of implementing a fix for that use case this project needed to be updated from its skeleton and modernized to align with how we do projects now.

## 🧪 Testing
Automated tests work as expected.
Edit:
I have tested this code on a FIPS system both installed as a library, and in standalone mode with Ansible as described in the README. There were no issues and I confirmed that it was successful in creating hashing objects with the `usedforsecurity` keyword argument.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]
These changes look good to me.  Have they been tested against a FIPS system?

File: src/ioc_scan/ioc_scanner.py
Comment (hillaryj) [inline src/ioc_scan/ioc_scanner.py]: Nit: should be `privileged`

File: tests/test_ioc_scan.py
Comment (hillaryj) [inline tests/test_ioc_scan.py:14]: Is it necessary to import both `ioc_scan` and specific components of it?

**Review by hillaryj — APPROVED** [review]
👍  Left some minor comments

File: src/ioc_scan/ioc_scanner.py
Comment (mcdonnnj) [reply src/ioc_scan/ioc_scanner.py]: Good catch! Fixed in 0f54aa1

File: tests/test_ioc_scan.py
Comment (mcdonnnj) [reply tests/test_ioc_scan.py:14]: The `ioc_scan` package doesn't export those component so the `import ioc_scan` is just to get `ioc_scan.__version__`. We need to import the specific components to access them for testing.

**Review by hillaryj — APPROVED** [review]
👍

File: tests/test_ioc_scan.py
Comment (hillaryj) [reply tests/test_ioc_scan.py:14]: Roger that!

File: .github/CODEOWNERS
Comment (dav3r) [inline .github/CODEOWNERS]: @mcdonnnj - please remove @hillaryj and @jsf9k as code owners, as part of our effort to be more targeted with owners.

File: bump_version.sh
Comment (dav3r) [inline bump_version.sh]: `example` --> `ioc_scan`

File: src/ioc_scan/ioc_scanner.py
Comment (dav3r) [inline src/ioc_scan/ioc_scanner.py]: Typo: `shoudl` --> `should`

**Review by dav3r — APPROVED** [review]
This is looking solid @mcdonnnj - I just noted a few minor thangs.  🦘

File: .github/CODEOWNERS
Comment (mcdonnnj) [reply .github/CODEOWNERS]: Should we also remove `@cisagov/team-ois`?

File: src/ioc_scan/ioc_scanner.py
Comment (mcdonnnj) [reply src/ioc_scan/ioc_scanner.py]: Oops. Fixed in 64f0e9d

File: bump_version.sh
Comment (mcdonnnj) [reply bump_version.sh]: Ah, good catch. Fixed in 48a4862

File: .github/CODEOWNERS
Comment (dav3r) [reply .github/CODEOWNERS]: Yes, please remove `@cisagov/team-ois` too - thanks!

Comment (mcdonnnj) [reply .github/CODEOWNERS]: 👍 Trimmed in 4480be7

**Comment by mcdonnnj** [comment]
> 
> 
> These changes look good to me. Have they been tested against a FIPS system?

@jsf9k I'm waiting for access to the test server (which is a FIPS system). Once I have confirmed functionality, I will merge unless there are outstanding change requests.

**Pull request merged**
